etcdctl/command/util.go

```diff
@@ -286,10 +290,10 @@ func contextWithTotalTimeout(c *cli.Context) (context.Context, context.CancelFun
 // syncWithPeerAPI syncs cluster with peer API defined at
 // https://github.com/coreos/etcd/blob/v0.4.9/server/server.go#L311.
 // This exists for backward compatibility with etcd 0.4.x.
-func syncWithPeerAPI(c *cli.Context, ctx context.Context, knownPeers []string) ([]string, error) {
-	tr, err := getTransport(c)
-	if err != nil {
-		return nil, err
+func syncWithPeerAPI(c *cli.Context, ctx context.Context, knownPeers []string) (eps []string, trerr error, serr error) {
```

Comment:
If it's pointed at a server that doesn't understand the request or the read is abnormally terminated, it should give "failed to connect" instead of giving a 400 error? Is "failed to connect to host" a failure to connect to an etcd server or is it a failure to connect to a server in general?